src/operator/controllers/secrets/manager.go

```diff
@@ -144,12 +157,15 @@ func (m *KubernetesSecretsManager) updateTLSSecret(ctx context.Context, config s
 		metadata.JKSPasswordAnnotation:                    config.CertConfig.JKSConfig.Password,
 		metadata.CertTypeAnnotation:                       string(config.CertConfig.CertType),
 	}
+	if config.ShouldRestartPodOnRenewal {
+		secret.Annotations[metadata.ShouldRestartOnRenewalAnnotation] = "remove this annotation to disable"
```

Comment:
Is this how we expect users to interact? The annotation that controls it should be on the pod, not the secret, right? This annotation being added is internal, so removing it might not lead to the desired result - the operator might re-add when it next updates the Secret, if the pod annotations still say to add it (if `config.ShouldRestartPodOnRenewal`)

Reply:
I will add "form the pod" to the text here. 
The code already checks if secret configuration changed like that: pod annotations parsed into secretConfig object, if the secret already exists we parse its annotations to secretConfig and check if it matches the pod secretConfig.